### ISSUE TITLE
Unexplained behavior "Input correct length"

### ISSUE THREAD

**Issue by scifood** [phpmyadmin/phpmyadmin#12731]
Unexplained behavior while editing comments only.

### Data
```sql
CREATE TABLE `cl$_shopping_lists` (
  `shopping_list_id` int(10) UNSIGNED NOT NULL COMMENT 'ID of shopping list',
  `user_id` int(10) UNSIGNED NOT NULL COMMENT 'ID of user',
  `date_add` int(10) UNSIGNED NOT NULL COMMENT 'Addition date of the shopping list (unsigned unix-time)',
  `date_update` int(10) UNSIGNED DEFAULT NULL COMMENT 'Update date of the shopping list (unsigned unix-time)',
  `shopping_list_name` varchar(255) DEFAULT NULL COMMENT 'Custom name of shopping list (defined by user)'
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COMMENT='Users'' shopping lists';

ALTER TABLE `cl$_shopping_lists`
  ADD PRIMARY KEY (`shopping_list_id`);

ALTER TABLE `cl$_shopping_lists`
  MODIFY `shopping_list_id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'ID of shopping list';
```
### Steps to reproduce
1. In "Structure" mode select "date_add","date_update","shopping_list_name" for change
2. Press "Preview SQL"
3.1. (first way): Instead message "No change" you'll see real SQL for updating
3.2. (second way): You'll see "No change", and after changing ONLY comments for this fields and saving you'll see "Input correct length"

**Comment by nijel** [comment]
What phpMyAdmin version do you use? I can not reproduce this with latest 4.6.5.1.

**Comment by Enverex** [comment]
I'm getting this when trying to create tables too. Using "VARCHAR" doesn't work, it simply always says "Input correct length" regardless of what you set length to. If you set the type to TEXT, save, go back in, edit the field to VARCHAR and set a length, it then works. Looks like the sanity checking is broken somewhere.

**Comment by ibennetch** [comment]
@enverex what phpMyAdmin version are you using?

**Comment by Enverex** [comment]
Sorry, thought I put that in the post before, it's 4.6.5.2.

**Comment by lem9** [comment]
Here a way to better see the missing "No change" problem, as tested in 4.8.1. Click on Structure, then for date_add click on Change, change nothing then click on Preview SQL: you get the "No change" message. Repeat for date_update and you get a generated statement, although you changed nothing.

If there are really two issues in this report, I suggest to split it.

**Comment by williamdes** [comment]
Confirmed.

**Comment by williamdes** [comment]
This seems fixed in 5.1 versions.
